### DISCUSSION TITLE
Connecting WSL to Windows

### DISCUSSION THREAD

**Discussion by DrBailey27** [Cyfrin/foundry-full-course-cu#920] [Q&A] [answered]
I am having issues connecting VSC to WSL on my windows
this is the error message I am getting
"No WSL distros found. New distros can be installed from the Microsoft Store."

can someone please help me.?

**Comment by 0xusmanf** [comment, marked as answer]
Could you try the following fixes, the error you mentioned typically happens when there are no Linux distributions installed on the Windows Subsystem for Linux (WSL).
1. Restart your machine after installing WSL.
2. Check if you have installed WSL correctly by running `wsl --version` in the command line. Below is an example output of the correctly installed WSL.
```shell
C:\Users\usman>wsl --version
WSL version: 1.2.5.0
Kernel version: 5.15.90.1
WSLg version: 1.0.51
MSRDC version: 1.2.3770
Direct3D version: 1.608.2-61064218
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.19045.3448
```
3. Check if you have installed the correct version of Linux by running the command `wsl --status` in the command line, below is an example output of the correct version.
 ```shell
C:\Users\usman>wsl --status
Default Distribution: Ubuntu
Default Version: 2

C:\Users\usman>
```
4. If it's installed correctly then run the WSL by running `wsl` command in the command line before you try to connect VS code to WSL, below is an example output when you run WSL.
```shell
C:\Users\usman>wsl
Welcome to Ubuntu 22.04.2 LTS (GNU/Linux 5.15.90.1-microsoft-standard-WSL2 x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage


This message is shown once a day. To disable it please create the
/home/usmanfarooq90/.hushlogin file.
usmanfarooq90@DESKTOP-8OOB3I3:/mnt/c/Users/usman$
```

**Reply by DrBailey27** [reply]
Thank you very much, it is working
